File: 023/merge-k-sorted-lists.js
var mergeKLists = function (lists) {
    var ans = [];

    lists.forEach(function (item) {
        while (item) {
            ans.push(new ListNode(item.val));
            item = item.next;
        }
    });

    ans.sort(function (a, b) {
        return a.val - b.val;
    });

    if (!ans.length)
        return null;

    for (var i = 0, len = ans.length - 1; i < len; i++)
        ans[i].next = ans[i + 1];

    return ans[0];
};

File: 023/MergeKSortedLists.scala
/**
 * Definition for singly-linked list.
 * class ListNode(var _x: Int = 0) {
 *   var next: ListNode = null
 *   var x: Int = _x
 * }
 */
object MergeKSortedLists {
    def mergeKLists(lists: Array[ListNode]): ListNode = {
        val q = collection.mutable.PriorityQueue.empty[ListNode](Ordering.by[ListNode, Int](_.x).reverse)
        lists.foreach(list => if (list != null) q += list)
        val res = new ListNode(-1)
        var p = res
        while (q.nonEmpty) {
            val head = q.dequeue()
            p.next = head
            p = p.next
            if (head.next != null)
                q += head.next
            head.next = null
        }
        res.next
    }
}